amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -0,0 +1,219 @@
+                    if (it.name.endsWith('transform.conf.json')) {
+                        transformConfFile = project.file(it)
+                    }
+                }
+            }
+            if (transformConfFile.isFile()) {
+                def tranformConfText = transformConfFile.text
+                def jsonSlurper = new groovy.json.JsonSlurper()
+                def transformConfJson = jsonSlurper.parseText(tranformConfText)
+
+                def resolverConfigMap = ['ResolverConfig':
+                                                 ['project':
+                                                          ['ConflictHandler'  : 'AUTOMERGE',
+                                                           'ConflictDetection': 'VERSION']
+                                                 ]
+                ]
+                if (!syncEnabled) {
+                    transformConfJson.remove('ResolverConfig')
+                } else if (!transformConfJson.ResolverConfig) {
+                    transformConfJson << resolverConfigMap
+                }
+                def transformConfJsonStr = groovy.json.JsonOutput.toJson(transformConfJson)
+                def transformConfJsonStrPretty = groovy.json.JsonOutput.prettyPrint(transformConfJsonStr)
+                transformConfFile.write(transformConfJsonStrPretty)
+            }
+        }
+
+        project.task('modelgen') {
+            doLast {
+                project.exec {
+                    commandLine 'amplify', 'codegen', 'model'
+                }
+            }
+        }
+
+        project.task('amplifyPush') {
+            def AWSCLOUDFORMATIONCONFIG
+            if (!accessKeyId || !secretAccessKey || !region) {
+                AWSCLOUDFORMATIONCONFIG = """{\
+\"configLevel\":\"project\",\
+\"useProfile\":true,\
+\"profileName\":\"$profile\"\
+}"""
+            } else {
+                AWSCLOUDFORMATIONCONFIG = """{\
+\"configLevel\":\"project\",\
+\"useProfile\":true,\
+\"profileName\":\"$profile\",\
+\"accessKeyId\":\"$accessKeyId\",\
+\"secretAccessKey\":\"$secretAccessKey\",\
+\"region\":\"$region\"\
+}"""
+            }
+
+            def AMPLIFY
+            if (!envName) {
+                AMPLIFY = """{\
+\"envName\":\"amplify\"\
+}"""
+            } else {
+                AMPLIFY = """{\
+\"envName\":\"$envName\"\
+}"""
+            }
+
+            def PROVIDERS = """{\
+\"awscloudformation\":$AWSCLOUDFORMATIONCONFIG\
+}"""
+
+            doLast {
+                def doesLocalEnvExist = project.file('./amplify/.config/local-env-info.json').exists()
+                if (doesLocalEnvExist) {
+                    project.exec {
+                        commandLine 'amplify', 'push', '--yes'
+                    }
+                } else {
+                    project.exec {
+                        commandLine 'amplify', 'init', '--amplify', AMPLIFY, '--providers', PROVIDERS, '--yes'
```

Comment:
The `PROVIDERS` variable has been configured with credentials above, and now we're passing those on the command line, where they will be visible to anyone who can view the command line. Can this command read `providers` from a file instead?

Reply:
This command line execution is not piped to STDOUT or any other output so it's not visible on the commandline. But if its better practice to use a file we can make this change

Reply:
Simply executing the command in a shell makes it visible to utilities like `ps`, as well as potentially other process accounting, logging, and auditing systems. Let's use a file. We've been using a protected home directory: `$HOME/.aws-amplify` with `700` permissions as the root repository for these well-known paths. I'd suggest something like `$HOME/.aws-amplify/amplify-android` as a reasonable containing directory for this file, but I'll let @jamesonwilliams have the final say.

Reply:
Yes. We should read from a file. Anything on the command line is available system wide, as @palpatim mentioned. I'm a little fuzzy and want to know more about why we need to transit credentials anywhere as they are available to the CLI already.

Reply:
These credentials are not available to the CLI because they're being pulled from the user editing amplify-gradle-config.json file, which is the CLI-less flow we worked on for this plugin. From my understanding, piping the PROVIDERS to the command line is no less secure than having the customer manually enter their secret key and access key to the command line when prompted by the CLI, which is an accepted flow that we already use.